wfe2/wfe.go

```diff
@@ -1573,64 +1573,63 @@ func (wfe *WebFrontEndImpl) Certificate(ctx context.Context, logEvent *web.Reque
 		return
 	}
 
-	leafPEM := pem.EncodeToMemory(&pem.Block{
-		Type:  "CERTIFICATE",
-		Bytes: cert.DER,
-	})
+	responsePEM, prob := func() ([]byte, *probs.ProblemDetails) {
+		leafPEM := pem.EncodeToMemory(&pem.Block{
+			Type:  "CERTIFICATE",
+			Bytes: cert.DER,
+		})
 
-	var responsePEM []byte
+		// If we don't have any certificateChains configured, just return the cert.
+		if len(wfe.certificateChains) == 0 {
+			return leafPEM, nil
+		}
```

Comment:
Is this case needed anymore? I think we now use this field consistently in prod and can make it a construct-time error for WFE to not have certificateChains.

Reply:
Agreed, I'll do that in a follow-up change.